### PR TITLE
Coveo search integration

### PR DESCRIPTION
**Pull request by nitin-rachabathuni**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-340, https://jira.corp.adobe.com/browse/EXLM-17

Test URLs:

Before: https://main--exlm--adobe-experience-league.hlx.live/en/browse
After: https://feature-exlm-340--exlm--adobe-experience-league.hlx.live/en/browse

### CODE REVIEW

File: blocks/header/header.js [hunk 1]
```diff
@@ -3,6 +3,9 @@ import { loadIms } from '../../scripts/scripts.js';
 import { signOut } from '../../scripts/auth/auth-operations.js';
 import Search from '../../scripts/search/search.js';
 import { registerResizeHandler } from './header-utils.js';
+// import coveoSearchEnginePOC from '../../scripts/search/coveo-headless-poc.js';
```
Comment (Sivaramvt) [inline blocks/header/header.js:6]: Let's remove these

File: scripts/coveo/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// copy folder recursively from ./node_modules/@coveo/headless/dist/browser to ./browser
```
Comment (Sivaramvt) [inline scripts/coveo/index.js:1]: Shall we rename this folder as `scripts/coveo-headless-libs`

File: blocks/browse-filters/browse-filters.js [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+    if (dropdownEl && !isCurrentDropDownOpen) {
+      dropdownEl.querySelector('.filter-dropdown-content').style.display = 'block';
+      dropdownEl.classList.add('open');
+    } else {
+      closeOpenDropdowns();
+    }
+  });
+}
+
+function decorateBlockTitle(block) {
```
Comment (Sivaramvt) [inline blocks/browse-filters/browse-filters.js:396]: This title is not part of the story. We may have to remove this

File: scripts/search/search.js [hunk 4]
```diff
@@ -2,6 +2,19 @@ import { htmlToElement, loadIms } from '../scripts.js';
 import SearchDelegate from './search-delegate.js';
 import { searchUrl } from '../urls.js';
 
+// eslint-disable-next-line import/no-relative-packages
+// import { buildSearchEngine } from '../coveo/browser/headless.esm.js';
+// console.log('buildSearchEngine', buildSearchEngine);
+
+// or via dymanic import
+setTimeout(async () => {
```
Comment (Sivaramvt) [inline scripts/search/search.js:10]: I suppose this search is used in header and gets loaded in all the page.
We may want to load the coveo-headless library only in browse-* pages

File: scripts/search/coveo-headless-poc.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
+import buildHeadlessSearchEngine from '../data-service/coveo/headless-engine.js';
```
Comment (Sivaramvt) [inline scripts/search/coveo-headless-poc.js:1]: file name - pls rename

File: scripts/search/search-engine.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import loadCoveoToken from '../data-service/coveo/coveo-token-service.js';
```
Comment (Sivaramvt) [inline scripts/search/search-engine.js:1]: Pls remove this

File: scripts/search/coveo-headless-poc.js (hunk 5 above)
Comment (Sivaramvt) [reply scripts/search/coveo-headless-poc.js:1]: let's move this to scripts/coveo-headless/?

File: scripts/data-service/coveo/headless-engine.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+export default function buildHeadlessSearchEngine(module, coveoToken) {
```
Comment (Sivaramvt) [inline scripts/data-service/coveo/headless-engine.js:1]: Move this to scripts/coveo-headless/?

Scripts / Coveo-headless - Our logic
Scripts / Coveo-headless/libs - Library

File: scripts/coveo/package.json [hunk 8]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (Sivaramvt) [inline scripts/coveo/package.json:1]: See if it's required

File: scripts/browse-card/browse-cards-coveo-data-adaptor.js [hunk 9]
```diff
@@ -44,10 +44,19 @@ const BrowseCardsCoveoDataAdaptor = (() => {
   const mapResultToCardsDataModel = (result) => {
     const { raw, parentResult, title, excerpt, uri } = result || {};
     /* eslint-disable camelcase */
-    const { el_contenttype, el_product } = parentResult?.raw || raw || {};
-    const contentType = Array.isArray(el_contenttype) ? el_contenttype[0]?.trim() : el_contenttype?.trim();
-    const product = Array.isArray(el_product) ? el_product[0] : el_product;
-    /* eslint-enable camelcase */
+    const { el_contenttype, el_product, el_solution, el_type } = raw || {};
```
Comment (Sivaramvt) [inline scripts/browse-card/browse-cards-coveo-data-adaptor.js:47]: const { el_contenttype, el_product } = parentResult?.raw || raw || {};

File: blocks/browse-filters/browse-filters.js (hunk 3 above)
Comment (thomsebastin) [reply blocks/browse-filters/browse-filters.js:396]: I have asked to keep title for now as removing them is no more rendering this as a block.  We will have a small task to follow up on this to not have title. Hope that works.

File: scripts/search/search.js [hunk 10]
```diff
@@ -2,6 +2,15 @@ import { htmlToElement, loadIms } from '../scripts.js';
...
 
+// or via dymanic import
+setTimeout(async () => {
+  // eslint-disable-next-line import/no-relative-packages
+  import('../coveo-headless/libs/browser/headless.esm.js').then((module) => {
+    // eslint-disable-next-line no-console
+    console.log('hey look! it loaded!', module.buildSearchEngine);
+  });
+}, 1000);
+
```
Comment (noormd08) [inline scripts/search/search.js:13]: As discussed, please remove the code from here and add it in browse-* file.

**Review by thomsebastin — APPROVED** [review]
@nitin-rachabathuni have noticed some minor bugs. Please follow up as discussed.

**Pull request merged**